### ISSUE TITLE
Fisheye View Coordinates

### ISSUE THREAD

**Issue by zwall-bp** [radiantlab/HDRICalibrationTool#33]
> As a user, I want to upload a camera's fisheye coordinates of an image, in order to accurately portray the area on the raw image which we will be manipulating.

# Background

This is a subtask of #7.

After setting up the response function, in order to ensure the HDR calculations are done only within the world, and not any blank spaces caused by the fisheye effect, the fisheye view coordinates need to be given and processed to crop the HDR image.

This step is required, as there is a void space in the raw input images that can severely affect the calculation of the HDR image.

More information about this step can be found in the [HDR Image Generation Tutorial](https://www.tandfonline.com/doi/full/10.1080/15502724.2019.1684319). See figure 1.

# Requirements

* The app will be a single page application. In the third section of the page, beneath the Response Function selection, there will be an input area for selecting the fisheye coordinates configuration.
* The user can click the button to open a file dialog window that will allow them to browse files and select a fisheye view coordinates file.
* The image's fisheye coordinates will be stored as a React state.
* The user can click a button to modify or remove the uploaded fisheye coordinates if desired.
* The file name (or perhaps full file path?) of the selected fisheye coordinates is displayed.

# Design Resources

There isn't a design resource yet in order to visualize the process of configuring the fisheye coordinates.

The following need to be completed before this issue:

#18

Before being able to run the hdrgen step of the pipeline, the following feature is also required:

#19

# Subtasks

This is a sister task to #28, with needing to input in the LDR image's configuration.

# Estimate

1 week.
# Open Questions

Is there a way to easily parse the image's metadata in order to get this information, instead of needing the user to manually enter it?

# Acceptance Criteria

1. The user opens the app.
1. The user selects the `Upload Fisheye View Coordinates` button.
1. The OS' file selection dialog is opened.
1. The name of the file selected from the dialog for the fisheye coordinates is displayed.
1. The user can click the Remove button to remove the uploaded file.
1. The name of the new uploaded file is displayed under the fisheye coordinates section.

Stretch goal: Drag and drop capabilities. Maybe automating the process through LDR image parsing?

# Definition of Done

The definition of done can be found within our [Software Development Process document](https://docs.google.com/document/u/1/d/1b9ew4qtVZ9rGZiTvdeFDgCCmq4kLCokuLqpRv2HTGIM/edit).
